homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,318 @@
+            container = await hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+        # pylint: disable=c-extension-no-member
+        except (av.error.FileNotFoundError):
+            return {CONF_STREAM_SOURCE: "stream_file_not_found"}, None, None
+        except (av.error.HTTPNotFoundError):
+            return {CONF_STREAM_SOURCE: "stream_http_not_found"}, None, None
+        except (av.error.TimeoutError):
+            return {CONF_STREAM_SOURCE: "timeout"}, None, None
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return {CONF_STREAM_SOURCE: "stream_unauthorised"}, None, None
+        except (KeyError, IndexError):
+            return {CONF_STREAM_SOURCE: "stream_novideo"}, None, None
+        except PermissionError:
+            return {CONF_STREAM_SOURCE: "stream_not_permitted"}, None, None
+        except OSError as err:
+            if err.errno == EHOSTUNREACH:
+                return {CONF_STREAM_SOURCE: "stream_no_route_to_host"}, None, None
+            if err.errno == EIO:  # input/output error
+                return {CONF_STREAM_SOURCE: "stream_io_error"}, None, None
+            raise err
+    name = info.get(CONF_NAME, url or stream_source)
+    return errors, fmt, name
+
+
+class GenericIPCamConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> GenericOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return GenericOptionsFlowHandler(config_entry)
+
+    def check_for_existing(self, options):
+        """Check whether an existing entry is using the same URLs."""
+        return any(
+            entry.options[CONF_STILL_IMAGE_URL] == options[CONF_STILL_IMAGE_URL]
+            and entry.options[CONF_STREAM_SOURCE] == options[CONF_STREAM_SOURCE]
+            for entry in self._async_current_entries()
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the start of the config flow."""
+        errors = {}
+        if user_input:
+            # Secondary validation because serialised vol can't seem to handle this complexity:
+            if not user_input.get(CONF_STILL_IMAGE_URL) and not user_input.get(
+                CONF_STREAM_SOURCE
+            ):
+                errors["base"] = "no_still_image_or_stream_url"
+            else:
+                (errors, still_format, name) = await async_test_connection(
+                    self.hass, user_input
+                )
+                if not errors and name:
```

Comment:
We should validate name right away since its a cheap operations.  Make it an error if the name is missing/invalid so your validation function checks right away and we don't have the complexity of checking both errors and name

Reply:
Solved by ensuring there's always a valid name.